test/conv_common.hpp

```diff
@@ -1486,13 +1534,48 @@ struct conv_driver : test_driver
 
     void run()
     {
-        filter.spatialDim       = get_spatial_dim();
+        filter.spatialDim       = spatial_dim_elements.size();
         filter.mode             = cmode_lookup[miopen::ToUpper(conv_mode)];
         filter.paddingMode      = pmode_lookup[miopen::ToUpper(pad_mode)];
         std::size_t spatial_dim = filter.GetSpatialDimension();
+        if(spatial_dim == 2)
+        {
+            input = tensor<T>{
+                batch_size,
+                input_channels,
+                spatial_dim_elements.at(0),
+                spatial_dim_elements.at(
+                    1)}.generate(tensor_elem_gen_integer{17});
+            weights = tensor<T>{
+                output_channels,
+                input_channels,
+                spatial_dim_elements.at(0),
+                spatial_dim_elements.at(
+                    1)}.generate(tensor_elem_gen_integer{17});
+        }
+        else if(spatial_dim == 3)
+        {
+            input = tensor<T>{
+                batch_size,
+                input_channels,
+                spatial_dim_elements.at(0),
+                spatial_dim_elements.at(1),
+                spatial_dim_elements.at(
+                    2)}.generate(tensor_elem_gen_integer{17});
+            weights = tensor<T>{
+                output_channels,
+                input_channels,
+                spatial_dim_elements.at(0),
+                spatial_dim_elements.at(1),
+                spatial_dim_elements.at(
+                    2)}.generate(tensor_elem_gen_integer{17});
+        }
+        else
+        {
+            MIOPEN_LOG_E("dimension is wrong!");
+        }
```

Comment:
This breaks passing in the configs with `--input` and `--weights`, which means the deepbench test_conv_extra tests are not working correctly. Also, it makes it harder to pass the config in from the log. 

It also makes it harder to control what cases we want to run.

Reply:
I plan on updating to use --input and --weights as overrides to the explicit batch-channel-filter usage. I think these only make it harder if you don't like putting in more than 2 flags, other than that it is pretty clear what is happening. In any case they users should be able to fall back to the original inputs.

Reply:
Yea, but you still have no way of avoiding too many cross product combinations or duplicating config we are already testing. 

You could use a for loop:

```cpp
std::vector<std::vector<std::size_t>> get_inputs()
{
    std::vector<std::vector<std::size_t>> result;
    for(auto batch_size:get_batch_sizes())
        for(auto batch_size:get_input_channels())
            for(auto spatial_dim_elements:get_spatial_dims())
            {
                std::vector<std::size_t> dim = {batch_size, input_channels};
                dim.insert(dim.end(), spatial_dim_elements.begin(), spatial_dim_elements.end());
                result.push_back(dim);
            }
    return result;
}
```

Although, I think explicitly listing them is better so we can better coverage with less tests.